### DISCUSSION TITLE
[Node Operator Question] Node instance running out of space

### DISCUSSION THREAD

**Discussion by Rogalek** [ethereum-optimism/developers#346] [Node Operators] [answered]
### Are you running the most up to date node software?

- [X] Yes

### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hello, I started running optimism mainnet node recently (few weeks ago) it is still synchronizing.
It is full node without legacy created from source.

I have a question how big its gonna grow? In the documentation there was an info that it should be around 1TB but I have already:
`Usage of /:   94.1% of 1.42TB` and its still growing.

Am I doing something wrong? Please help.

### Protocol Description

tags:
optimism: op-dispute-mon/v0.1.2-1-g786f19907
op-geth: v1.101308.3-rc.1-8-g336d284b6

network: Mainnet
chainid: 10
l1 chain: Eth

**Comment by hash1go** [comment, marked as answer]
Judging by the size of [recent snapshots](https://datapoint.fastnode.io/) provided by third-party services, it will likely use more than 1.5TB of SSD capacity.
If the documentation is outdated, we will consider updating it.

**Reply by sbvegan** [reply]
Yeah we just need to update the docs, its hard to keep up to date because the blockchain is always growing. I'll note that you're using odd versions of the `op-node` and `op-geth`. You can find that [here](https://docs.optimism.io/builders/node-operators/releases)

**Reply by Rogalek** [reply]
yes it is outdated: https://docs.optimism.io/builders/node-operators/tutorials/mainnet#hardware-requirements

> Hardware Requirements
Hardware requirements for OP Mainnet nodes can vary depending on the type of node you plan to run. Archive nodes generally require significantly more resources than full nodes. Below are suggested minimum hardware requirements for each type of node.

>16GB RAM
1TB SSD (full node) or 4TB SSD (archive node)
Reasonably modern CPU

I was updating op-node and op-geth just after ETH update, did it changed already? Ok thanks I will update it.

I think it will be nice also section in documentation where you specify how exactly should we update op-node and op-geth repos if you are using them from source, we need run all build steps again right?

**Reply by Chomtana** [reply]
I think you could try snap sync from scratch if you want to reduce disk usage temporarily (Not sure).

Snap sync docs: https://docs.optimism.io/builders/node-operators/management/snap-sync

I don't know deeply how snap sync work. However, blockchain always grow in size, there is currently no way to prune efficiently.

**Reply by Chomtana** [reply]
> I think it will be nice also section in documentation where you specify how exactly should we update op-node and op-geth repos if you are using them from source, we need run all build steps again right?

The essential step is `make op-node`.

I don't sure if you need to run pnpm commands but it doesn't seem neccessary.
